src/Plugin.Maui.Audio/AudioPlayer/AudioPlayer.android.cs

```diff
@@ -7,198 +7,226 @@ namespace Plugin.Maui.Audio;
 
 partial class AudioPlayer : IAudioPlayer
 {
-    readonly MediaPlayer player;
-    int index = 0;
-    double volume = 0.5;
-    double balance = 0;
-    string path = string.Empty;
-    readonly MemoryStream? stream;
-    bool isDisposed = false;
-
-    public double Duration => player.Duration / 1000.0;
-
-    public double CurrentPosition => player.CurrentPosition / 1000.0;
-
-    public double Volume
-    {
-        get => volume;
-        set => SetVolume(volume = value, Balance);
-    }
-
-    public double Balance
-    {
-        get => balance;
-        set => SetVolume(Volume, balance = value);
-    }
-
-    public bool IsPlaying => player.IsPlaying;
-
-    public bool Loop
-    {
-        get => player.Looping;
-        set => player.Looping = value;
-    }
+   readonly MediaPlayer player;
+   int index = 0;
+   double volume = 0.5;
+   double balance = 0;
+   string path = string.Empty;
+   readonly MemoryStream? stream;
+   bool isDisposed = false;
+
+   IDispatcherTimer? myTimer = null;
+   DateTime startTime;
+   public TimeSpan Ts = TimeSpan.Zero;
+
+   public double Duration {
+      get {
+         double duration = Ts.TotalMilliseconds/1000;
+         if (duration == 0)
+            duration = player.Duration / 1000.0;
+         return duration;
+      }
+   }
+
+   void t_Tick(object? sender, EventArgs e)
+   {
+      Ts = DateTime.Now - startTime;
+   }
```

Comment:
I am not sure I fully follow what this is trying to achieve? Why couldn't we just expose `player.Duration`. If we do go ahead I don't think this is be in the scope of this PR though